apps/moderation/serializers.py

```diff
@@ -3,17 +3,18 @@
 from rest_framework import serializers
 from rest_framework.serializers import ModelSerializer
 
-from apps.moderation.models.moderated_model import ModeratedModel
 
+def get_moderated_model_serializer(
+    model_serializer: Type[ModelSerializer],
+) -> Type[ModelSerializer]:
+    """Return the serializer to be used for moderation of a moderated model.
+    Adds 'fields' field to given model_serializer.
+    """
 
-def get_moderated_model_serializer(model_cls: Type[ModeratedModel]) -> Type[ModelSerializer]:
-    """Return the serializer to be used for moderation of a moderated model."""
-
-    class ModeratedModelSerializer(ModelSerializer):
+    class ModeratedModelSerializer(model_serializer):
         fields = serializers.ReadOnlyField(source='get_moderated_fields')
 
-        class Meta:
-            model = model_cls
-            include = model_cls.get_moderated_fields()
+        class Meta(model_serializer.Meta):
+            fields = model_serializer.Meta.fields + ['fields']
```

Comment:
@alashow, what do you think about this?

```
        class Meta(model_serializer.Meta):
            fields = model_serializer.Meta.model.get_moderated_fields() + ['fields']
```
I'm considering the possibility of `model_serializer.Meta.fields` being different than `model_serializer.Meta.model.get_moderated_fields()`.

Reply:
I'm not sure how would that work?
If we do that, `get_moderated_model_serializer` won't have any original fields. It will have "moderated fields" in `fields` field + an array of "moderated fields" ( which is a dict, moderatad field names as key and field info as value, and not compatible with `fields` array)

Reply:
Oh you're totally right. I was thinking that `get_moderated_fields` would return a list of strings (field names).

Reply:
Since this serializer is for moderation, should we account for `model_serializer.Meta.model.Moderation.excluded_fields`? Maybe something like ...
```
fields = [
    field for field in model_serializer.Meta.fields 
    if field not in model_serializer.Meta.model.Moderation.excluded_fields
] + ['fields']
```

Reply:
I don't know why we would want to exclude fields for moderation? Do you have an example? 
`get_moderated_model_serializer` should just help extend normal serializers to add moderated fields info. 

Also, I haven't decided to use the function for moderation yet because we would want something that works with `ModelViewSet`s.

Reply:
Actually, if we use `fields` for building the moderation form, then it's probably OK to not worry about limiting the fields in the serializer.  So, never mind.

But to answer your question, the idea of excluding fields for moderation came from the django-moderation package. And here are the default excluded fields for `ModeratedModel`:
```
class Moderation:
        excluded_fields = [
            'changes',
            'verified',
            'deleted',
            'cache',
            'pretty_cache',
            'date_string',
        ]
```